saucerbot/the_dores.py

```diff
@@ -0,0 +1,134 @@
+        for team in teams:
+            if team['team']['location'] == 'Vanderbilt':
+                return ev
+    logger.info("Looked through all the events, couldn't find the Vandy game")
+    return None
+
+
+def __get_teams(game):
+    """
+    Extracts the teams from the game. Vandy will be returned first
+    :param game: the game information
+    :return: both teams, vandy first
+    """
+    team1 = game['competitions'][0]['competitors'][0]
+    team2 = game['competitions'][0]['competitors'][1]
+    if team1['team']['location'] == 'Vanderbilt':
+        return team1, team2
+    else:
+        return team2, team1
+
+
+def __get_week(desired_date):
+    """
+    We're gonna assume that week 1 is always Labor Day  (which, in older seasons before like 2007 wasn't the
+    case, so we may need a better method for this in the future)
+    :param desired_date: the date we're checking the week from
+    :return: the week number
+    """
+    labor_day = datetime.datetime(desired_date.year, 9, 1)
+    while labor_day.weekday() != 0:
+        labor_day = labor_day + datetime.timedelta(days=1)
+    week_1 = labor_day - datetime.timedelta(4)  # we'll say Thursday is the one we want to calculate from
+    diff = desired_date - week_1
+    week = diff.days / 7 + 1
+    return week
+
+
+def did_the_dores_win(print_in_progress=False, print_loss=False, desired_date=None):
+    """
+    Checks if the dores won on the desired date! It'll return a response in the case of a win, or a loss with the
+    first argument set to true. Right now it only does football, but basketball should be pretty easy to incorporate
+    :param print_loss: if True, will return a message in the event of a loss. If false, None will be returned for a loss
+    :param desired_date: the date to check the score from
+    :return: A String message if Vandy won, then either None or a losing message if we lost, depending on the parameter
+    """
+    if desired_date is None:
+        desired_date = datetime.datetime.now()
+    game = get_football_results(desired_date)
+    if game is None:
+        logger.debug("No game found")
+        return None
+    else:
+        vandy, opponent = __get_teams(game)
+        if game['status']['type']['completed'] is False:
+            logger.debug("Game still in progress")
+            if print_in_progress:
+                return random.choice(IN_PROGRESS_MESSAGES)
+            else:
+                return None
+        elif vandy['winner'] is True:
+            response = random.choice(WINNING_FORMATS)
+        else:
+            if print_loss:
+                response = random.choice(LOSING_FORMATS)
+            else:
+                logger.debug("Vandy lost, but no printing is enabled")
+                return None
+        return response.format(team=opponent['team']['displayName'], vandy_score=vandy['score'],
+                               other_score=opponent['score'])
+
+
+WINNING_FORMATS = [
```

Comment:
I think these might need to be at the top of the module?  I can't remember exactly if this will work or not lol